### PR TITLE
Wallet cooldown

### PR DESCRIPTION
**Pull request by maikelmclauflin**
fixes https://github.com/brave-intl/internal/issues/105

### CODE REVIEW

File: ledger/controllers/grants.js [hunk 1]
```diff
@@ -1161,3 +1169,14 @@ function uploadTypedGrants (protocolVersion, uploadSchema, contentSchema) {
     reply({})
   }
 }
+
+function cooldownOffset () {
+  console.log('cooldown calc', typeof WALLET_COOLDOWN_HRS, WALLET_COOLDOWN_HRS)
```
Comment (nvonpentz) [inline ledger/controllers/grants.js]: nit: log statement

File: ledger/controllers/grants.js [hunk 2]
```diff
@@ -1161,3 +1169,14 @@ function uploadTypedGrants (protocolVersion, uploadSchema, contentSchema) {
...
+  const hours = WALLET_COOLDOWN_HRS ? (+WALLET_COOLDOWN_HRS || 0) : 24
+  return hours * 60 * 60 * 1000
+}
+
+function walletCooldownId () {
+  const hexString = braveUtils.hexTime(-cooldownOffset())
+  return bson.ObjectId(hexString)
+}
```
Comment (nvonpentz) [inline ledger/controllers/grants.js]: could we unit test walletCooldownId and cooldownOffset?

File: bat-utils/lib/extras-utils.js [hunk 3]
```diff
@@ -48,6 +49,12 @@ function documentOlderThan (olderThanDays, anchorTime, _id) {
   return createdTimestamp(_id) < (anchorTime - (DAY_MS * olderThanDays))
 }
 
+function hexTime (offset = 0, now = new Date()) {
+  const offsetMS = now.getTime() - offset
+  const offsetSeconds = Math.floor(offsetMS / 1000)
+  return offsetSeconds.toString(16) + '0000000000000000'
+}
+
```
Comment (nvonpentz) [inline bat-utils/lib/extras-utils.js]: could we unit test hexTime?

**Comment by maikelmclauflin** [comment]
> Adding an integration test would be great too if time permits

so i thought about this, i just couldn't think of any way to do it unless we want to create some sort of bypass token that we can use in tests. happy to do that if wanted

File: bat-utils/lib/extras-utils.js [hunk 4]
```diff
@@ -48,6 +49,11 @@ function documentOlderThan (olderThanDays, anchorTime, _id) {
   return createdTimestamp(_id) < (anchorTime - (DAY_MS * olderThanDays))
 }
 
+function hexTime (now = new Date()) {
```
Comment (evq) [inline bat-utils/lib/extras-utils.js]: I would prefer we call this something like `timestampToMongoId` or similar

Comment (maikelmclauflin) [reply bat-utils/lib/extras-utils.js]: can do

Comment (maikelmclauflin) [reply bat-utils/lib/extras-utils.js]: just realized it's not even used anymore, so removing

**Pull request merged**